### PR TITLE
PSR formatting and type hinting. respects fully qualified class names.

### PR DESCRIPTION
**Pull request by TomK**
PSR code formatting, use fully qualified class names to preserve reflection functionality. Supersedes #80

### CODE REVIEW

File: lib/PayPal/Auth/Oauth/AuthSignature.php [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
+        $response      = $authSignature->genSign($key, $secret, $token, $tokenSecret, $httpMethod, $endpoint);
+        return "token=" . $token .
+        ",signature=" . $response['oauth_signature'] .
+        ",timestamp=" . $response['oauth_timestamp'];
+    }
+
 }
-?>
\ No newline at end of file
+
```
Comment (jaypatel512) [inline lib/PayPal/Auth/Oauth/AuthSignature.php]: Extra newline

**Comment by TomK** [comment]
Hi Jay, most other files already have a newline at end of file, for the reasons outlined in many answers here: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

I'm happy to remove them if you need, but its pretty standard to end all lines with /n or /r/n.

File: lib/PayPal/Auth/Oauth/AuthSignature.php [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
+    public static function generateFullAuthString($key, $secret, $token, $tokenSecret, $httpMethod, $endpoint)
+    {
+        $authSignature = new AuthSignature();
+        $response      = $authSignature->genSign($key, $secret, $token, $tokenSecret, $httpMethod, $endpoint);
+        return "token=" . $token .
+        ",signature=" . $response['oauth_signature'] .
+        ",timestamp=" . $response['oauth_timestamp'];
+    }
+
```
Comment (jaypatel512) [inline lib/PayPal/Auth/Oauth/AuthSignature.php]: Extra newline here. My mistake.

**Comment by jaypatel512** [comment]
Oh ! I added the comment at the wrong place. corrected now.

**Comment by TomK** [comment]
Ah i see, yeah that is because the formatter is configured to leave a blank line before php close tags. I will remove the close tags completely as they are not recommended anyway.

**Comment by TomK** [comment]
@jaypatel512 let me know if there are further changes required.

**Comment by jaypatel512** [comment]
This PR looks good. I will go ahead and merge them. Thanks again @TomK for your wonderful help. Really appreciated.

**Pull request merged**